### ISSUE TITLE
In testing stage, the input of the used network is different from the input of training stage.

### ISSUE THREAD

**Issue by TL9792** [nwuAI/CAN#1]

**Comment by TL9792** [comment]
Hello, for the released code by you, I found the input of the network is different from the  input of the testing stage when reproducing your segmentation result, i.e., the input has image_3D,

**Comment by TL9792** [comment]
Hello, for the released code by you, I found the input of the network is different from the input of the testing stage when reproducing your segmentation result, i.e., there are image_3D,skull_3D,img_2D in the training stage, while there are only image_3D in the testing stage.  The released code is right?

**Comment by nwuAI** [comment]
hi, TL9792. Thank you for your question.
We uploaded an incorrect version earlier and we have now corrected it. Please see the file test.py, test_coarse.py and test_fine.py.
The detailed codes you should be found below

-------------------------------------------------------------------------------------------------------------------------

                    image_2D=volume[i,:,:][np.newaxis, :, :]
                    image_2D=image_2D.unsqueeze(0)

                    if i <= int(self.args.num_slices * 2):
                        image_stack01 = volume[0:int(i), :, :][None]
                        image_stack02 = volume[int(i+1):int(self.args.num_slices * 2 + 1), :, :][None]
                        skull_stack01 = skull[0:int(i), :, :][None]
                        skull_stack02 = skull[int(i+1):int(self.args.num_slices * 2 + 1), :, :][None]
                    elif i==int(self.args.num_slices * 2 + 1):
                        image_stack01 = volume[0:int(i-1), :, :][None]
                        image_stack02 = volume[int(i + 1):int(self.args.num_slices * 2 + 1), :, :][None]
                        skull_stack01 = skull[0:int(i-1), :, :][None]
                        skull_stack02 = skull[int(i + 1):int(self.args.num_slices * 2 + 1), :, :][None]
                    elif i >= 245 and i< 255:
                        image_stack01 = volume[z_ax - int(self.args.num_slices * 2 + 1):int(i), :, :][None]
                        image_stack02 = volume[int(i+1):, :, :][None]
                        skull_stack01 = skull[z_ax - int(self.args.num_slices * 2 + 1):int(i), :, :][None]
                        skull_stack02 = skull[int(i+1):, :, :][None]
                    elif i == 255:
                        image_stack01 = volume[z_ax - int(self.args.num_slices * 2 + 1):-1, :, :][None]
                        image_stack02 = volume[i+1:, :, :][None]
                        skull_stack01 = skull[z_ax - int(self.args.num_slices * 2 + 1):-1, :, :][None]
                        skull_stack02 = skull[i+1:, :, :][None]
                    else:
                        image_stack01 = volume[i - self.args.num_slices:i, :, :][None]
                        image_stack02 = volume[i+1 :i+self.args.num_slices+1, :, :][None]
                        skull_stack01 = skull[i - self.args.num_slices:i, :, :][None]
                        skull_stack02 = skull[i+1 :i+self.args.num_slices+1, :, :][None]

                    image_3D = torch.cat((image_stack01, image_stack02), dim=1)
                    skull_3D=torch.cat((skull_stack01,skull_stack02),dim=1)

                    outputs = self.model(image_3D,skull_3D,image_2D)

-------------------------------------------------------------------------------------------------------------------------
